### PR TITLE
CONSOLE-2522: Migrate CRD CRUD tests to Cypress

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Fixes https://issues.redhat.com/browse/CONSOLE-2522.

### CODE REVIEW

**Comment by rebeccaalpert** [comment]
Hey @dtaylor113 - could you review this when you get a chance? Thanks!

File: frontend/packages/integration-tests-cypress/tests/crud/customresourcedefinition.spec.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import { listPage } from '../../views/list-page';
+import * as yamlEditor from '../../views/yaml-editor';
+import { errorMessage } from '../../views/form';
+
+describe('CustomResourceDefinitions', () => {
+  const plural = `crd${testName}`;
+  const group = 'test.example.com';
+  const name = `${plural}.${group}`;
+  const testLabel = 'automatedTestName';
+  const crd = {
```
Comment (spadgett) [inline frontend/packages/integration-tests-cypress/tests/crud/customresourcedefinition.spec.ts:15]: Not new, but would be good to add the type

```suggestion
  const crd: CustomResourceDefinitionKind = {
```

File: frontend/packages/integration-tests-cypress/tests/crud/customresourcedefinition.spec.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+  it('creates, displays, and deletes `CustomResourceDefinitions` and creates a new custom resource instance', () => {
+    it('displays a YAML editor for creating a new custom resource definition', () => {
+      cy.visit('/k8s/cluster/customresourcedefinitions');
+      listPage.rows.shouldBeLoaded();
+      listPage.clickCreateYAMLbutton();
+      yamlEditor.isLoaded();
+      let newContent;
+      yamlEditor.getEditorContent().then((content) => {
+        newContent = _.defaultsDeep({}, crd, safeLoad(content));
```
Comment (spadgett) [inline frontend/packages/integration-tests-cypress/tests/crud/customresourcedefinition.spec.ts:98]:
```suggestion
      yamlEditor.getEditorContent().then((content) => {
        const newContent = _.defaultsDeep({}, crd, safeLoad(content));
```

File: frontend/packages/integration-tests-cypress/tests/crud/customresourcedefinition.spec.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        });
+      });
+      cy.visit(`/k8s/cluster/customresourcedefinitions?name=${name}`);
+      listPage.rows.shouldBeLoaded();
+      listPage.rows.clickKebabAction(`CRD${testName}`, 'View instances');
+      listPage.clickCreateYAMLbutton();
+      yamlEditor.isLoaded();
+      let newContent;
+      yamlEditor.getEditorContent().then((content) => {
+        newContent = _.defaultsDeep({}, customResource, safeLoad(content));
```
Comment (spadgett) [inline frontend/packages/integration-tests-cypress/tests/crud/customresourcedefinition.spec.ts:111]:
```suggestion
      yamlEditor.getEditorContent().then((content) => {
        const newContent = _.defaultsDeep({}, customResource, safeLoad(content));
```

File: frontend/packages/integration-tests-cypress/tests/crud/customresourcedefinition.spec.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        yamlEditor.setEditorContent(safeDump(newContent, { sortKeys: true })).then(() => {
+          yamlEditor.clickSaveCreateButton();
+          cy.get(errorMessage).should('not.exist');
+        });
+      });
+      cy.visit(`/k8s/cluster/customresourcedefinitions?name=${name}`);
+      listPage.rows.shouldBeLoaded();
+      listPage.rows.clickKebabAction(`CRD${testName}`, 'Delete CustomResourceDefinition');
```
Comment (spadgett) [inline frontend/packages/integration-tests-cypress/tests/crud/customresourcedefinition.spec.ts:119]: We should confirm this succeeded somehow.

**Review by dtaylor113 — APPROVED** [review]
/lgtm  continue to love how readable we've been able to make Cypress with our helper view objs.

**Comment by rebeccaalpert** [comment]
Added missing import.

**Pull request merged**